src/bin/pgcopydb/snapshot.c

```diff
@@ -29,8 +29,7 @@ copydb_copy_snapshot(CopyDataSpec *specs, TransactionSnapshot *snapshot)
 
 	/* remember if the replication slot has been created already */
 	snapshot->exportedCreateSlotSnapshot = source->exportedCreateSlotSnapshot;
-
-	strlcpy(snapshot->pguri, source->pguri, sizeof(snapshot->pguri));
+	snapshot->pguri = strdup(source->pguri);
```

Comment:
Then I suppose we need to call `free(snapshot->pguri)` in the function that closes the snapshot?